### PR TITLE
Remove direct imports from react and use @wordpress/data

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue #4535 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @hussain-t. Looks like something still relies on react in our codebase. Could you please check e2e tests and figure out why they are failing? 

```
● Tag Manager module setup › Setup without AMP active › displays account creation form when user has no Tag Manager account

    ReferenceError: React is not defined

      at Object.241 (../../http:/localhost:9002/wp-content/plugins/google-site-kit/dist/assets/js/googlesitekit-settings-3d0b4a76f8daf3f281d4.min.js:15:7934)
      at __webpack_require__ (../../http:/localhost:9002/wp-content/plugins/google-site-kit/dist/assets/js/runtime-1010c1e63fefce33ed56.min.js:1:615)
      at Module.741 (../../http:/localhost:9002/wp-content/plugins/google-site-kit/dist/assets/js/googlesitekit-settings-3d0b4a76f8daf3f281d4.min.js:38:919)
      at __webpack_require__ (../../http:/localhost:9002/wp-content/plugins/google-site-kit/dist/assets/js/runtime-1010c1e63fefce33ed56.min.js:1:615)
      at _ (../../http:/localhost:9002/wp-content/plugins/google-site-kit/dist/assets/js/runtime-1010c1e63fefce33ed56.min.js:1:430)
      at Array.r [as push] (../../http:/localhost:9002/wp-content/plugins/google-site-kit/dist/assets/js/runtime-1010c1e63fefce33ed56.min.js:1:293)
      at ../../http:/localhost:9002/wp-content/plugins/google-site-kit/dist/assets/js/googlesitekit-settings-3d0b4a76f8daf3f281d4.min.js:1:79
      at Suite.<anonymous> (specs/modules/tagmanager/setup.test.js:129:3)
```

**Comment by hussain-t** [comment]
> Thanks, @hussain-t. Looks like something still relies on react in our codebase. Could you please check e2e tests and figure out why they are failing?

Thanks @eugene-manuilov. For some reason when I globally searched for `from 'react``, VS Code was not showing the recently updated two other files before pushing the initial commit. When I restarted VS Code those were shown.

I also executed the e2e tests after the fix all are passing 👍

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Ah.. ok, but looks like something is still missing there. Can you take another look? Build your assets before running e2e tests with `npm run build:test` command because this is what we use in our workflows.

**Comment by hussain-t** [comment]
> Ah.. ok, but looks like something is still missing there. Can you take another look? Build your assets before running e2e tests with npm run build:test command because this is what we use in our workflows.

So, we need the `ProvidePlugin` but for `@wordpress/element` module like so:

```javascript
new ProvidePlugin( {
  React: '@wordpress/element',
} )
```

**Comment by hussain-t** [comment]
@eugene-manuilov 
Update: We need the `ProvidePlugin` for `@wordpress/element` module in the `.storybook/webpack.config.js` file as well.

**Review by eugene-manuilov — APPROVED** [review]
LGTM! Thanks, @hussain-t.

File: webpack.config.js [hunk 1]
```diff
@@ -272,7 +272,7 @@ function* webpackConfig( env, argv ) {
 		},
 		plugins: [
 			new ProvidePlugin( {
-				React: 'react',
+				React: '@wordpress/element',
```
Comment (tofumatt) [inline webpack.config.js:275]: Is this needed? I removed it and everything still built fine. It's minor, and I know it _is_ needed in the Jest tests and in Storybook, but maybe we should remove it here.

Comment (tofumatt) [reply webpack.config.js:275]: Nevermind! I see why it's required now; all good 👍🏻

Comment (hussain-t) [reply webpack.config.js:275]: Yes, it is needed for e2e and VRT tests 👍

**Pull request merged**